src/operations/client_bulk_write/results_merger.ts

```diff
@@ -84,18 +81,84 @@ export class ClientBulkWriteResultsMerger {
             if (document.upserted) {
               result.upsertedId = document.upserted._id;
             }
-            this.result.updateResults?.set(document.idx + currentBatchOffset, result);
+            this.result.updateResults?.set(document.idx + this.currentBatchOffset, result);
           }
           // Handle delete results.
           if ('delete' in operation) {
-            this.result.deleteResults?.set(document.idx + currentBatchOffset, {
+            this.result.deleteResults?.set(document.idx + this.currentBatchOffset, {
               deletedCount: document.n
             });
           }
+        } else {
```

Comment:
I think the if statement above needs to be adjusted - otherwise we'll enter the `else` block every time for any top-level error we receive where verbose results is false.

I think what we want is:
```typescript
if (document.ok === 1) {
	if (this.options.verboseResults) {
		...
	}
} else {
	
}

```

Reply:
Updated, good catch.